docs/extended-tour.md

```diff
@@ -0,0 +1,159 @@
+# Extended Tour
+Welcome to the extended tour of node-cpp-skel. This documentation is especially handy if you're just starting out with building [Node Addon-ons](https://github.com/mapbox/cpp/blob/master/node-cpp.md) or writing C++ :tada:
+
+## Table of Contents:
+- [Walkthrough of example code](extended-tour.md#walkthrough-example-code)
+- [What does "build" mean?](extended-tour.md#what-does-build-mean)
+- [Software build components](extended-tour.md#software-build-components)
+- [Configuration files](extended-tour.md#configuration-files)
+- [Autogenerated files](extended-tour.md#autogenerated-files)
+- [Overall flow](extended-tour.md#overall-flow)
+
+# Walkthrough example code
+
+This skeleton includes a few examples of how you might design your application, including standalone functions and creating Objects/Classes. Both the synchronous and asynchronous versions of each are included to give an iterative example of what the progression from sync to async looks like. Let's run through reasons why you'd design your code in these ways:
+1. [Standalone function](API.md#hello-1)
+2. [Standalone asynchronous function](../API.md#hello_async)
+3. [Object/Class](../API.md#helloobject)
+4. [Asynchronous Object/Class](../API.md#helloobjectasync)
+
+
+### When would you use a standalone function?
+Standalone functions are simple and useful when your code is doing some kind of basic operation. They avoid the need for a class, since creating a class in this situation would add unnecessary complexity. Creating an asynchronous standalone function can be powerful if your code has a specific, yet expensive operation to do and would benefit from the use of multiple threads at one time to do that expensive work.
```

Comment:
How about:

> A standalone function is a function that exists at the top level of the module scope rather than as a member of an object that is instantiated. So if your module is `wonderful` a standalone function would be called like `wonderful.callme()`. A standalone function makes sense when the only data needed by the function can be easily passed as arguments. When it is not easy or clean to pass data as arguments then you should consider encapsulation, an example of which would be exposing a function as a member of an object.

In reality you could achieve clean encapulation either by passing a class/object as an argument to a standalone function (as long as there are public accessor methods to get access to the right data) or by exposing a member function on that class/object that has access to the data. Which is chosen is a matter of preference - both would be equally high performance as long as the class involved pre-calculates/pre-creates as much as possible to be efficient (https://github.com/mapbox/cpp/blob/master/glossary.md#efficiency)